src/bio/bio_monitor.c

```diff
@@ -811,8 +837,16 @@ bio_bs_monitor(struct bio_xs_context *xs_ctxt, uint64_t now)
 		D_ERROR("State transition on target %d failed. %d\n",
 			bbs->bb_owner_xs->bxc_tgt_id, rc);
 
-	if (!bypass_health_collect())
-		collect_raw_health_data(xs_ctxt);
+	if (!bypass_health_collect()) {
+		D_ALLOC_PTR(gsca);
```

Comment:
Instead of allocating and freeing a 'gsca', can't we just pass '&xs_ctxt->bxc_xs_blobstore[st]" as argument for completion callback?